src/Stateless/DotGraph/DotGraphFormatter.cs

```diff
@@ -7,78 +7,237 @@ namespace Stateless.DotGraph
     /// <summary>
     /// DOT GraphViz text writer for reflection API.
     /// </summary>
-    public static class DotGraphFormatter
+    public class DotGraphFormatter
     {
+        Dictionary<string, string> clusterDictionary;
+        List<string> _handledStates;
+        List<string> _formattedNodesList;
+        FormatList _nodeShapeState;
+        FormatList _clusterFormat;
+        FormatList _edgeShape;
+        FormatList _exitEdges;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public DotGraphFormatter()
+        {
+            _nodeShapeState = new FormatList()
+                .Add(new Shape(ShapeType.Mrecord))
+                .Add(new Color(HtmlColor.blue));
+            _clusterFormat = new FormatList()
+                .Add(new Label("placeholder"));
+            _edgeShape = new FormatList()
+                .Add(new Label("placeholder"))
+                .Add(new Style(ShapeStyle.solid));
+            _exitEdges = new FormatList()
+                .Add(new Label("placeholder"))
+                .Add(new Style(ShapeStyle.dotted));
+        }
+
+        // TODO: Is this the way we want to create DOT graphs?  new DotGraphFormatter().ToDotGraph(sm)?
+
         /// <summary>
-        /// Produces a DOT GraphViz graph.
+        /// A string representation of the stateRepresentation machine in the DOT graph language.
         /// </summary>
-        /// <param name="stateMachineInfo">The StateMachineInfo to be mapped.</param>
-        /// <returns>DOT GraphViz text.</returns>
-        public static string Format(StateMachineInfo stateMachineInfo)
+        /// <returns>A description of all simple source states, triggers and destination states.</returns>
+        public string ToDotGraph(StateMachineInfo machineInfo)
         {
-            var bindings = stateMachineInfo.States;
+            clusterDictionary = new Dictionary<string, string>();
+            _handledStates = new List<string>();
+            _formattedNodesList = new List<string>();
```

Comment:
The three assignments here could be joined with the declaration, saving some clutter.